lib/Support/Config/TargetConfig.cpp

```diff
@@ -0,0 +1,338 @@
+    output << "if [ -f \"${install_dir}/lib/libnvqir-"
+           << configValue.SimulationBackend.values.front() << ".so\" ]; then\n";
+    output << "  NVQIR_SIMULATION_BACKEND=\""
+           << configValue.SimulationBackend.values.front() << "\"\n";
+    // If there are more than one simulator libs, create the `else` paths to
+    // check their .so files.
+    for (std::size_t i = 1; i < configValue.SimulationBackend.values.size();
+         ++i) {
+      output << "elif [ -f \"${install_dir}/lib/libnvqir-"
+             << configValue.SimulationBackend.values[i] << ".so\" ]; then\n";
+      output << "  NVQIR_SIMULATION_BACKEND=\""
+             << configValue.SimulationBackend.values[i] << "\"\n";
+    }
+    output << "else\n";
+    output << "  error_exit=\"Unable to find NVQIR simulator lib for target "
+           << targetName << ". Please check your installation.\"\n";
+    output << "fi\n";
+  }
+
+  for (const auto &rule : configValue.ConditionalBuildConfigs) {
+    output << "if [[ " << rule.Condition << " ]]; then\n";
+    if (!rule.CompileFlag.empty()) {
+      output << "  COMPILER_FLAGS=\"${COMPILER_FLAGS} " << rule.CompileFlag
+             << "\"\n";
+    }
+    if (!rule.LinkFlag.empty()) {
+      output << "  LINKER_FLAGS=\"${LINKER_FLAGS} " << rule.LinkFlag << "\"\n";
+    }
+    output << "fi\n";
+  }
+
+  return output.str();
+}
+
+std::string
+cudaq::config::processRuntimeArgs(const cudaq::config::TargetConfig &config,
+                                  const std::vector<std::string> &targetArgv) {
+  std::stringstream output;
+  if (config.BackendConfig.has_value()) {
+    output << processSimBackendConfig(config.Name,
+                                      config.BackendConfig.value());
+  }
+
+  unsigned featureFlag = 0;
+  std::stringstream platformExtraArgs;
+  for (std::size_t idx = 0; idx < targetArgv.size();) {
+    const auto argsStr = targetArgv[idx];
+    const auto iter = std::find_if(
+        config.TargetArguments.begin(), config.TargetArguments.end(),
+        [&](const cudaq::config::TargetArgument &argConfig) {
+          // Here, we handle both cases: the config key as is or prefixed with
+          // the target name.
+          const std::string nvqppArgKey =
+              "--" + config.Name + "-" + argConfig.KeyName;
+          return (nvqppArgKey == argsStr) || (argsStr == argConfig.KeyName);
+        });
+    if (iter != config.TargetArguments.end()) {
+      if (iter->Type != cudaq::config::ArgumentType::FeatureFlag) {
+        // If this is a platform option (platform argument key is provide),
+        // forward the value to the platform extra arguments.
+        if (!iter->PlatformArgKey.empty()) {
+          platformExtraArgs << ";" << iter->PlatformArgKey << ";"
+                            << targetArgv[idx + 1];
+        }
+      } else {
+        // This is an option flag, construct the value for mapping selection.
+        const auto featureFlags = targetArgv[idx + 1];
```

Comment:
Same here. Perhaps even add a new `else` branch to throw an error.
```suggestion
      } else if (idx + 1 < targetArgv.size()) {
        // This is an option flag, construct the value for mapping selection.
        const auto featureFlags = targetArgv[idx + 1];
```